From: Luigi Gangitano <luigi@debian.org>
Date: Sat, 29 Oct 2016 17:55:02 +0200
Subject: Default configuration file for debian

---
 src/cf.data.pre | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/cf.data.pre b/src/cf.data.pre
index 588e0f81..c1356475 100644
--- a/src/cf.data.pre
+++ b/src/cf.data.pre
@@ -1693,11 +1693,12 @@ http_access deny manager
 #
 # INSERT YOUR OWN RULE(S) HERE TO ALLOW ACCESS FROM YOUR CLIENTS
 #
+include /etc/squid/conf.d/*.conf
 
 # Example rule allowing access from your local networks.
 # Adapt localnet in the ACL section to list your (internal) IP networks
 # from where browsing should be allowed
-http_access allow localnet
+#http_access allow localnet
 http_access allow localhost
 
 # And finally deny all other access to this proxy
@@ -4815,7 +4816,7 @@ DOC_END
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the default number of logfile rotations to make when you
@@ -4839,6 +4840,8 @@ DOC_START
 	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 	<pid>'.
 
+	Note, for Debian/Linux the default of logfile_rotate is
+	zero, since it includes external logfile-rotation methods.
 DOC_END
 
 NAME: mime_table
