From: Amos Jeffries <amosjeffries@squid-cache.org>
Last-Update: 2023-06-19
Subject: Default configuration file for debian
Forwarded: not-needed

---
 src/cf.data.pre | 4 +++-
 1 file changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/cf.data.pre b/src/cf.data.pre
index 46b87c47a..10788a1d8 100644
--- a/src/cf.data.pre
+++ b/src/cf.data.pre
@@ -1877,6 +1877,7 @@ http_access deny to_linklocal
 #
 # INSERT YOUR OWN RULE(S) HERE TO ALLOW ACCESS FROM YOUR CLIENTS
 #
+include /etc/squid/conf.d/*.conf
 
 # For example, to allow access from your local networks, you may uncomment the
 # following rule (and/or add rules that match your definition of "local"):
@@ -5313,7 +5314,7 @@ DOC_END
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the default number of logfile rotations to make when you
@@ -5337,6 +5338,8 @@ DOC_START
 	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 	<pid>'.
 
+	Note, for Debian/Linux the default of logfile_rotate is
+	zero, since it includes external logfile-rotation methods.
 DOC_END
 
 NAME: mime_table
