From: Amos Jeffries <amosjeffries@squid-cache.org>
Last-Update: 2023-06-19
Subject: Default configuration file for debian
Forwarded: not-needed

---
 src/cf.data.pre | 4 +++-
 1 file changed, 3 insertions(+), 1 deletions(-)

Index: pkg-squid/src/cf.data.pre
===================================================================
--- pkg-squid.orig/src/cf.data.pre	2024-01-18 12:58:14.645002402 +0100
+++ pkg-squid/src/cf.data.pre	2024-01-18 12:58:14.641002402 +0100
@@ -1912,6 +1912,7 @@
 #
 # INSERT YOUR OWN RULE(S) HERE TO ALLOW ACCESS FROM YOUR CLIENTS
 #
+include /etc/squid/conf.d/*.conf
 
 # For example, to allow access from your local networks, you may uncomment the
 # following rule (and/or add rules that match your definition of "local"):
@@ -5453,7 +5454,7 @@
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the default number of logfile rotations to make when you
@@ -5477,6 +5478,8 @@
 	in the habit of using 'squid -k rotate' instead of 'kill -USR1
 	<pid>'.
 
+	Note, for Debian/Linux the default of logfile_rotate is
+	zero, since it includes external logfile-rotation methods.
 DOC_END
 
 NAME: mime_table
