From: Amos Jeffries <squid3@treenet.co.nz>
Date: Thu, 16 Sep 2021 03:10:00 +1300

diff --git a/src/icmp/Makefile.am b/src/icmp/Makefile.am
index 54fdaf6b..05db3750 100644
--- a/src/icmp/Makefile.am
+++ b/src/icmp/Makefile.am
@@ -96,16 +96,21 @@ SquidConfig.cc: $(top_srcdir)/src/SquidConfig.cc
 	cp $(top_srcdir)/src/SquidConfig.cc $@
 
 tests/stub_HelperChildConfig.cc: $(top_srcdir)/src/tests/stub_HelperChildConfig.cc
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/stub_HelperChildConfig.cc $@
 
 tests/stub_fd.cc: $(top_srcdir)/src/tests/stub_fd.cc
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/stub_fd.cc $@
 
 tests/stub_libmem.cc: $(top_srcdir)/src/tests/stub_libmem.cc
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/stub_libmem.cc $@
 
 tests/stub_SBuf.cc: $(top_srcdir)/src/tests/stub_SBuf.cc
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/stub_SBuf.cc $@
 
 tests/STUB.h: $(top_srcdir)/src/tests/STUB.h
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/STUB.h $@
diff --git a/tools/Makefile.am b/tools/Makefile.am
index d31e05a5..286bc6bf 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -32,6 +32,7 @@ test_tools.cc: $(top_srcdir)/test-suite/test_tools.cc
 	cp $(top_srcdir)/test-suite/test_tools.cc $@
 
 tests/stub_debug.cc: $(top_srcdir)/src/tests/stub_debug.cc
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/stub_debug.cc $@
 
 Here.cc: $(top_srcdir)/src/base/Here.cc
@@ -47,12 +48,15 @@ time.cc: $(top_srcdir)/src/time.cc
 	cp $(top_srcdir)/src/time.cc $@
 
 tests/stub_cbdata.cc: $(top_srcdir)/src/tests/stub_cbdata.cc
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/stub_cbdata.cc $@
 
 tests/stub_libmem.cc: $(top_srcdir)/src/tests/stub_libmem.cc
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/stub_libmem.cc $@
 
 tests/STUB.h: $(top_srcdir)/src/tests/STUB.h
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/STUB.h $@
 
 # stock tools for unit tests - library independent versions of dlink_list
diff --git a/tools/squidclient/Makefile.am b/tools/squidclient/Makefile.am
index 3a76c7dc..bca486b4 100644
--- a/tools/squidclient/Makefile.am
+++ b/tools/squidclient/Makefile.am
@@ -31,15 +31,18 @@ test_tools.cc: $(top_srcdir)/test-suite/test_tools.cc
 	cp $(top_srcdir)/test-suite/test_tools.cc $@
 
 tests/stub_debug.cc: $(top_srcdir)/src/tests/stub_debug.cc
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/stub_debug.cc $@
 
 time.cc: $(top_srcdir)/src/time.cc
 	cp $(top_srcdir)/src/time.cc $@
 
 tests/stub_libmem.cc: $(top_srcdir)/src/tests/stub_libmem.cc
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/stub_libmem.cc $@
 
 tests/STUB.h: $(top_srcdir)/src/tests/STUB.h
+	$(MKDIR) $(builddir)/tests || $(TRUE)
 	cp $(top_srcdir)/src/tests/STUB.h $@
 
 # stock tools for unit tests - library independent versions of dlink_list
