Index: pkg-squid/src/internal.cc
===================================================================
--- pkg-squid.orig/src/internal.cc	2024-01-18 12:58:59.829002423 +0100
+++ pkg-squid/src/internal.cc	2024-01-18 12:58:59.825002423 +0100
@@ -162,6 +162,8 @@
     return host;
 }
 
+// XXX: This check mishandles cases where the listening port hostname
+//      does not match the request-target hostname.
 int
 internalHostnameIs(const char *arg)
 {
@@ -174,6 +176,9 @@
         if (0 == strcmp(arg, w->key))
             return 1;
 
+    if (0 == strcmp(arg, "localhost"))
+        return 1;
+
     return 0;
 }
 
