Author: Luigi Gangitano <luigi@debian.org>
Description: Default configuration file for debian
--- a/src/cf.data.pre
+++ b/src/cf.data.pre
@@ -178,7 +178,7 @@
 	If you want to use the traditional NCSA proxy authentication, set
 	this line to something like
 
-	auth_param basic program @DEFAULT_PREFIX@/libexec/ncsa_auth @DEFAULT_PREFIX@/etc/passwd
+	auth_param basic program @DEFAULT_PREFIX@/lib/squid3/ncsa_auth @DEFAULT_PREFIX@/etc/passwd
 
 	"utf8" on|off
 	HTTP uses iso-latin-1 as characterset, while some authentication
@@ -246,7 +246,7 @@
 	If you want to use a digest authenticator, set this line to
 	something like
 
-	auth_param digest program @DEFAULT_PREFIX@/bin/digest_pw_auth @DEFAULT_PREFIX@/etc/digpass
+	auth_param digest program @DEFAULT_PREFIX@/lib/squid3/digest_pw_auth @DEFAULT_PREFIX@/etc/digpass
 
 	"utf8" on|off
 	HTTP uses iso-latin-1 as characterset, while some authentication
@@ -308,7 +308,7 @@
 	of type proxy_auth.  By default, the NTLM authenticator_program
 	is not used.
 
-	auth_param ntlm program @DEFAULT_PREFIX@/bin/ntlm_auth
+	auth_param ntlm program @DEFAULT_PREFIX@/lib/squid3/ntlm_auth
 
 	"children" numberofchildren
 	The number of authenticator processes to spawn (no default).
@@ -343,7 +343,7 @@
 	The only supported program for this role is the ntlm_auth
 	program distributed as part of Samba, version 4 or later.
 
-	auth_param negotiate program @DEFAULT_PREFIX@/bin/ntlm_auth --helper-protocol=gss-spnego
+	auth_param negotiate program @DEFAULT_PREFIX@/lib/squid3/ntlm_auth --helper-protocol=gss-spnego
 
 	"children" numberofchildren
 	The number of authenticator processes to spawn (no default).
@@ -760,11 +760,11 @@
 # Example rule allowing access from your local networks.
 # Adapt to list your (internal) IP networks from where browsing
 # should be allowed
-acl localnet src 10.0.0.0/8	# RFC1918 possible internal network
-acl localnet src 172.16.0.0/12	# RFC1918 possible internal network
-acl localnet src 192.168.0.0/16	# RFC1918 possible internal network
-acl localnet src fc00::/7       # RFC 4193 local private network range
-acl localnet src fe80::/10      # RFC 4291 link-local (directly plugged) machines
+#acl localnet src 10.0.0.0/8	# RFC1918 possible internal network
+#acl localnet src 172.16.0.0/12	# RFC1918 possible internal network
+#acl localnet src 192.168.0.0/16	# RFC1918 possible internal network
+#acl localnet src fc00::/7       # RFC 4193 local private network range
+#acl localnet src fe80::/10      # RFC 4291 link-local (directly plugged) machines
 
 acl SSL_ports port 443
 acl Safe_ports port 80		# http
@@ -923,7 +923,7 @@
 # Example rule allowing access from your local networks.
 # Adapt localnet in the ACL section to list your (internal) IP networks
 # from where browsing should be allowed
-http_access allow localnet
+#http_access allow localnet
 http_access allow localhost
 
 # And finally deny all other access to this proxy
@@ -1424,7 +1424,7 @@
 
 	Note: The use of this directive using client dependent ACLs is
 	incompatible with the use of server side persistent connections. To
-	ensure correct results it is best to set server_persisten_connections
+	ensure correct results it is best to set server_persistent_connections
 	to off when using this directive in such configurations.
 DOC_END
 
@@ -1527,7 +1527,7 @@
 	an additional ACL needs to be used which ensures the IPv6-bound traffic
 	is never forced or permitted out the IPv4 interface.
 
-	# IPv6 destination test along with a dummy access control to perofrm the required DNS
+	# IPv6 destination test along with a dummy access control to perform the required DNS
 	# This MUST be place before any ALLOW rules.
 	acl to_ipv6 dst ipv6
 	http_access deny ipv6 !all
@@ -2844,7 +2844,7 @@
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the number of logfile rotations to make when you
@@ -2863,6 +2863,9 @@
 
 	Note, from Squid-3.1 this option has no effect on the cache.log,
 	that log can be rotated separately by using debug_options
+
+ 	Note2, for Debian/Linux the default of logfile_rotate is
+ 	zero, since it includes external logfile-rotation methods.
 DOC_END
 
 NAME: emulate_httpd_log
@@ -4284,7 +4287,7 @@
 NAME: visible_hostname
 TYPE: string
 LOC: Config.visibleHostname
-DEFAULT: none
+DEFAULT: localhost
 DOC_START
 	If you want to present a special hostname in error messages, etc,
 	define this.  Otherwise, the return value of gethostname()
@@ -6482,8 +6485,8 @@
 
 	WARNING:
 	  This option will restrict the situations under which IPv6
-	  connectivity is used (and tested). Hiding network problems
-	  which would otherwise be detected and warned about.
+	  connectivity is used (and tested), potentially hiding network
+	  problem swhich would otherwise be detected and warned about.
 DOC_END
 
 NAME: ipcache_size
