Author: Luigi Gangitano <luigi@debian.org>
Description: Change default file locations for debian
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -2802,7 +2802,7 @@
 DEFAULT_PREFIX = $(prefix)
 DEFAULT_CONFIG_DIR = $(sysconfdir)
 DEFAULT_CONFIG_FILE = $(DEFAULT_CONFIG_DIR)/squid.conf
-DEFAULT_MIME_TABLE = $(DEFAULT_CONFIG_DIR)/mime.conf
+DEFAULT_MIME_TABLE = $(datadir)/mime.conf
 DEFAULT_DNSSERVER = $(libexecdir)/`echo dnsserver | sed '$(transform);s/$$/$(EXEEXT)/'`
 DEFAULT_SSL_CRTD = $(libexecdir)/`echo ssl_crtd  | sed '$(transform);s/$$/$(EXEEXT)/'`
 DEFAULT_LOG_PREFIX = $(DEFAULT_LOG_DIR)
