Author: Luigi Gangitano <luigi@debian.org>
Description: Fix path for cachemgr.cgi default configuration file
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -34,7 +34,7 @@
 man_MANS = \
 	squidclient.1
 
-DEFAULT_CACHEMGR_CONFIG = $(sysconfdir)/cachemgr.conf
+DEFAULT_CACHEMGR_CONFIG = /etc/squid/cachemgr.conf
 
 squidclient_SOURCES = squidclient.cc
 nodist_squidclient_SOURCES=time.cc stub_debug.cc
@@ -56,6 +56,7 @@
 $(OBJS): $(top_srcdir)/include/version.h ../include/autoconf.h
 
 install-data-local:
+	test -z "/etc/squid" || $(MKDIR_P) "$(DESTDIR)/etc/squid"
 	$(INSTALL_DATA) $(srcdir)/cachemgr.conf $(DESTDIR)$(DEFAULT_CACHEMGR_CONFIG).default
 	@if test -f $(DESTDIR)$(DEFAULT_CACHEMGR_CONFIG) ; then \
 	        echo "$@ will not overwrite existing $(DESTDIR)$(DEFAULT_CACHEMGR_CONFIG)" ; \
--- a/tools/Makefile.in
+++ b/tools/Makefile.in
@@ -375,7 +375,7 @@
 man_MANS = \
 	squidclient.1
 
-DEFAULT_CACHEMGR_CONFIG = $(sysconfdir)/cachemgr.conf
+DEFAULT_CACHEMGR_CONFIG = /etc/squid/cachemgr.conf
 squidclient_SOURCES = squidclient.cc
 nodist_squidclient_SOURCES = time.cc stub_debug.cc
 cachemgr__CGIEXT__SOURCES = cachemgr.cc
@@ -1094,6 +1094,7 @@
 $(OBJS): $(top_srcdir)/include/version.h ../include/autoconf.h
 
 install-data-local:
+	test -z "/etc/squid" || $(MKDIR_P) "$(DESTDIR)/etc/squid"
 	$(INSTALL_DATA) $(srcdir)/cachemgr.conf $(DESTDIR)$(DEFAULT_CACHEMGR_CONFIG).default
 	@if test -f $(DESTDIR)$(DEFAULT_CACHEMGR_CONFIG) ; then \
 	        echo "$@ will not overwrite existing $(DESTDIR)$(DEFAULT_CACHEMGR_CONFIG)" ; \
