Index: squid3-3.1.18/src/Store.h
===================================================================
--- squid3-3.1.18.orig/src/Store.h	2011-12-26 22:10:23.000000000 +0100
+++ squid3-3.1.18/src/Store.h	2011-12-26 22:33:09.000000000 +0100
@@ -191,7 +191,7 @@
 
 #if USE_ADAPTATION
     /// call back producer when more buffer space is available
-    void deferProducer(const AsyncCall::Pointer &producer);
+    void deferProducer(AsyncCall::Pointer &producer);
     /// calls back producer registered with deferProducer
     void kickProducer();
 #endif
Index: squid3-3.1.18/src/store.cc
===================================================================
--- squid3-3.1.18.orig/src/store.cc	2011-12-26 22:10:23.000000000 +0100
+++ squid3-3.1.18/src/store.cc	2011-12-26 22:33:09.000000000 +0100
@@ -368,7 +368,7 @@
 
 #if USE_ADAPTATION
 void
-StoreEntry::deferProducer(const AsyncCall::Pointer &producer)
+StoreEntry::deferProducer(AsyncCall::Pointer &producer)
 {
     if (!deferredProducer)
         deferredProducer = producer;
